### PR TITLE
Fixed Disabled ads tracking option is reverted after reloading

### PR DESCRIPTION
**Pull request by simonhong**
C++ gives ad control type as bool but webui compared it with string.

Resolves https://github.com/brave/brave-browser/issues/11585

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Review by antonok-edm — APPROVED** [review]
Good catch, thank you!

File: browser/resources/settings/default_brave_shields_page/default_brave_shields_page.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ Polymer({
 
     Promise.all([this.browserProxy_.getAdControlType(), this.browserProxy_.isFirstPartyCosmeticFilteringEnabled()])
         .then(([adControlType, hide1pContent]) => {
-      this.adControlType_ = (adControlType === 'allow' ? 'allow' : (hide1pContent ? 'block' : 'block_third_party'))
+      this.adControlType_ = (adControlType ? 'allow' : (hide1pContent ? 'block' : 'block_third_party'))
```
Comment (pes10k) [inline browser/resources/settings/default_brave_shields_page/default_brave_shields_page.js:75]: This looks great but nit, maybe would read cleaner to just not have nested ternary statements at all

**Review by bsclifton — APPROVED** [review]
LGTM ++

File: browser/resources/settings/default_brave_shields_page/default_brave_shields_page.js (hunk 1 above)
Comment (bsclifton) [reply browser/resources/settings/default_brave_shields_page/default_brave_shields_page.js:75]: ++ from me on that too

Comment (simonhong) [reply browser/resources/settings/default_brave_shields_page/default_brave_shields_page.js]: Changed to improve readablity. PTAL again :)

Comment (pes10k) [reply browser/resources/settings/default_brave_shields_page/default_brave_shields_page.js]: nailed it :)

**Comment by simonhong** [comment]
Let me know if we want to uplift this cc @rebron @bsclifton @kjozwiak

**Comment by bsclifton** [comment]
Yes - please do! Worst case, @brave/uplift-approvers can reject it 😄

**Comment by GeetaSarvadnya** [comment]
Verification is in-progress

```
Brave | 1.16.46 Chromium: 86.0.4240.55 (Official Build) nightly (64-bit)
-- | --
Revision | a6d625ef6f7fe8ea0675f1cf759155a05ee1be40-refs/branch-heads/4240@{#953}
OS | Windows 10 OS Version 1903 (Build 18362.1016)

```
- Verified the STR mentioned in the description https://github.com/brave/brave-browser/issues/11585#issue-692106975
- Verified the scenarios mentioned in https://github.com/brave/brave-browser/issues/11585#issuecomment-693582182

**Pull request merged**